waiter/src/waiter/state.clj

```diff
@@ -1170,11 +1170,13 @@
         (and has-unhealthy-instances? (= first-unhealthy-status 401)) :health-check-requires-authentication))))
 
 (defn get-instability-issue
-  "Returns true if 1 or more instances failed due to OOM"
+  "Returns appropriate instability issue (only oom for now) for a service based on its instances, or nil if no such issues."
   [failed-instances]
   (when (not-empty failed-instances)
```

Comment:
I found this easier to read:
```
(when (some #(contains? (:flags %) :memory-limit-exceeded) failed-instances-flags)
  :not-enough-memory)
```

We can use the `cond` version when we have more instability issues to support. At that stage, maybe we want to return a set of instability issues rather than a single one.

Reply:
Agreed! I was initially trying to make it extendable for other instability issues but this makes sense. Thanks.